doc/external_frame.md

```diff
@@ -0,0 +1,221 @@
+# External Frame API Guide
+
+## 1. Overview
+
+In the general API, the video frames used in MTL are allcated by library with rte_malloc(usually with hugepages).
+In some use cases, each frame data needs to be copied from/to user. When it's UHD or high framerate video, the copying can cause CPU/memory stall and become bottleneck of the whole pipeline.
+The external frame API is introduced so the library can use user provided memory to receive/send st2110-20 frames, or as the color format conversion destination/source in the pipeline API.
+
+## 2. ST20 Pipeline ext_frame API
+
+### 2.1 st_ext_frame
+
+```c
+/** The structure info for external frame */
+struct st_ext_frame {
+  /** Each plane's virtual address of external frame */
+  void* addr[ST_MAX_PLANES];
+  /** Each plane's IOVA of external frame */
+  mtl_iova_t iova[ST_MAX_PLANES];
+  /** Each plane's linesize of external frame, leave as 0 if no line padding */
```

Comment:
Seems linesize set to zero?

Reply:
> 

no, should we add the comment or explicitly handle the 0 linesize in st_pipeline library?

Reply:
Maybe comment is more better. Since many APIs are directly using linesize.